session/sshforward/sshprovider/pipe.go

```diff
@@ -0,0 +1,116 @@
+package sshprovider
+
+import (
+	"context"
+	"net"
+	"sync"
+)
+
+// pipeListener is a net.Listener that uses net.Pipe to create connections
```

Comment:
Looks like this is only used for tests. In that case, can we remove it to `_test.go` files as Go dead code elimination is known to be quite bad.